Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/AssetFolderThumbnailView.cpp

```diff
@@ -459,6 +459,12 @@ namespace AzQtComponents
         }
     }
 
+    void AssetFolderThumbnailView::RefreshRootIndex()
+    {
+        QAbstractItemView::setRootIndex(rootIndex());
+        emit rootIndexChanged(rootIndex());
+    }
+
```

Comment:
By this point, the model has already been updated, but the issue is that the thumbnail view doesn't know the model's been updated so `UpdateGeometries()` hasn't been called and a repaint isn't done. I tried manually calling UpdateGeometries(), but that still doesn't trigger a repaint until a viewport event happens.

The root index is the currently selected folder, whose children the thumbnail view displays. Changing the root index doesn't update the entire widget, it just triggers a repaint and calls `UpdateGeometries()`. It is the equivalent of clicking on the same folder to update the current view.

If there's a way to also manually trigger a repaint, I'd be happy to make the change and just call a repaint and `UpdateGeometries()` 😃

Reply:
Awesome thank you!  One last thing, should be able to use `update()` instead of `repaint()`. Here's some info on why it's preferred:

https://doc.qt.io/qt-6/qwidget.html#update

Reply:
Oh nice, updated!